File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/PensaoAlimenticiaJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import java.math.BigDecimal;
import java.util.Date;

public abstract class PensaoAlimenticiaJRAdapter
extends JRAdapter {
    public abstract Date getOcorrencia();

    public abstract String getFormula();

    public abstract BigDecimal getBase();

    public abstract BigDecimal getAliquota();

    public abstract BigDecimal getDevido();
}

File: src/lib/pjecalc/core/dominio/calculo/relatorio/pensao-alimenticia-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — PensaoAlimenticiaJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.PensaoAlimenticiaJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/PensaoAlimenticiaJRAdapter.java
 */
import type Decimal from 'decimal.js';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class PensaoAlimenticiaJRAdapter extends JRAdapter {
  abstract getAliquota(): Decimal;
  abstract getOcorrencias(): JRAdapterDataSource<PensaoOcorrenciaAdapter>;
  abstract getTotalDevido(): Decimal;
  abstract getTotalCorrigido(): Decimal;
}

export abstract class PensaoOcorrenciaAdapter extends JRAdapter {
  abstract getCompetencia(): Date;
  abstract getBase(): Decimal;
  abstract getValorPensao(): Decimal;
  abstract getIndiceCorrecao(): Decimal;
  abstract getValorCorrigido(): Decimal;
}
